CONTRIBUTING.md

```diff
@@ -0,0 +1,58 @@
+# Contributing Guidelines
+
+Thanks for thinking about contributing to Project Calico! The success of an
+open source project is entirely down to the efforts of its contributors, so we
+do genuinely want to thank you for even thinking of contributing.
+
+This document contains some guidance and steps for contributing. Make sure you
+read it thoroughly, because if you miss some of these steps we will have to ask
+you to do them after you make your submission, and that slows down our ability
+to merge it.
+
+## Before Contributing: The Contributor License Agreement
+
+If you plan to contribute in the form of documentation or code, before we can
+accept your contribution we need you to sign our Contributor License Agreement.
```

Comment:
This sounds slightly run-on when reading it, you can shorten it to something like:

We need you to sign out Contributor License Agreement before we can accept your code or documentation contribution to Project Calico.